### PR TITLE
Enhance shell app with index.html

### PR DESCRIPTION
**Pull request by hankchiutw**
Just clone the repo and execute `qbrt run shell` to have a hello world page.

### CODE REVIEW

**Comment by mykmelez** [comment]
Thanks for the contribution! I'm curious about how this differs from `qbrt run example`, which seems to do the same thing. I wonder if there's a way for the shell app to provide a different experience. Alternately, perhaps we don't need the example app with these changes to the shell?

**Comment by hankchiutw** [comment]
I think the benefit of using shell app is you can define some XUL specified features. And the question would be: what kind of features are XUL limited?(i.e. can not be implemented by js, html, jsm). But, I doubt how many developers are interested in XUL development. lol.

**Comment by mykmelez** [comment]
> I think the benefit of using shell app is you can define some XUL specified features. And the question would be: what kind of features are XUL limited?(i.e. can not be implemented by js, html, jsm). But, I doubt how many developers are interested in XUL development. lol.

Indeed, there may not be very many developers who are interested in building XUL apps. But it could still be valuable to demonstrate one. How about creating an examples/ directory and putting a new XUL app into it, calling it something like examples/hello-world-xul/? Then we could move the example/ directory to examples/hello-world-html/ and use the two directories to compare an HTML app to a XUL one!

(The reason I suggest a separate app, instead of reusing the shell app, is that the shell app is used internally to implement running/packaging of a URL, so I'd like to avoid including code in it that is just for demonstration purposes.)

**Comment by hankchiutw** [comment]
Understand. I update a commit, please have a look!

File: examples/hello-world-xul/main.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  Cc['@mozilla.org/widget/macdocksupport;1'].getService(Ci.nsIMacDockSupport).activateApplication(true);
+}
+
+const argument = Cc['@mozilla.org/supports-string;1'].createInstance(Ci.nsISupportsString);
+argument.data = WINDOW_URL;
+
+// TODO: report error if URL isn't found.
+const window = Services.ww.openWindow(null, SHELL_URL, '_blank', WINDOW_FEATURES, argument);
+Runtime.openDevTools(window);
+console.log('main.js loaded!');
```
Comment (mykmelez) [inline examples/hello-world-xul/main.js]: This is currently demonstrating how to build a XUL "shell" for an HTML app (i.e. the shell app). I would simplify it to focus on the use case of building a XUL app itself, i.e. to show the XUL equivalent of the hello-world HTML app.

Thus I would make main.js simply open a XUL window called index.xul, while index.xul would simply display "Hello, World!" in that window. Of course there are other features of XUL that this app could demonstrate, but simply displaying "Hello, World!" would be a good way to begin.

(It's also helpful for the two hello-world apps to have equivalent functionality, to the extent possible, so a user can look at them side-by-side to see the difference between developing in HTML vs. XUL. So if we add features to the XUL app in the future, then we should try to add the same features to the HTML app.)

**Comment by hankchiutw** [comment]
how about this?

**Comment by mykmelez** [comment]
> how about this?

It looks good! Two small nits, which I'll fix when merging:

1. According to the [XUL tutorial](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XUL/Tutorial/Adding_Labels_and_Images), "You cannot embed text directly into a XUL file without tags around it and expect it to be displayed. You can use two XUL elements for this purpose." Strangely, the "Hello, World!" text in index.xul does get displayed. But it'd be more correct to use a `<label>` element.
2. Empty elements in XUL like `<script src="app.js"></script>` are generally written as an empty-element tag (a.k.a. the "self-closing syntax"), i.e. `<script src="app.js"/>`.

**Comment by mykmelez** [comment]
Merged with those changes. Great work, thanks @hankchiutw!

Note: I'd be happy to see more example apps in the examples/ directory, as well as new features in the existing example apps. So if there's something you're interested in demonstrating, feel free to propose and/or implement it!

(For the "hello-world" apps, as I mentioned before, it would be ideal if the two apps had equivalent features, so it's possible to compare the implementation in HTML to the implementation in XUL. For other example apps, however, there's no need to implement two different versions. A single app, whether HTML or XUL, is just fine.)

**Comment by hankchiutw** [comment]
Thanks! Looking forward to seeing the project growing up!

**Pull request merged**